evennia/contrib/base_systems/character_creator/character_creator.py

```diff
@@ -0,0 +1,201 @@
+    you left off later.
+    """
+    key = "charcreate"
+    locks = "cmd:pperm(Player) and is_ooc()"
+    help_category = "General"
+
+    def func(self):
+        "create the new character"
+        account = self.account
+        session = self.session
+        
+        # only one character should be in progress at a time, so we check for WIPs first
+        in_progress = [chara for chara in account.db._playable_characters if chara.db.chargen_step]
+        
+        if len(in_progress):
+            # we're continuing chargen for a WIP character
+            new_character = in_progress[0]
+        else:
+            # we're making a new character
+            charmax = settings.MAX_NR_CHARACTERS if settings.MULTISESSION_MODE > 1 else 1
+            
+            if not account.is_superuser and (
+                account.db._playable_characters and len(account.db._playable_characters) >= charmax
+            ):
+                plural = "" if charmax == 1 else "s"
+                self.msg(f"You may only create a maximum of {charmax} character{plural}.")
+                return
+
+            # create the new character object, with default settings
+            start_location = ObjectDB.objects.get_id(settings.START_LOCATION)
+            default_home = ObjectDB.objects.get_id(settings.DEFAULT_HOME)
+            permissions = settings.PERMISSION_ACCOUNT_DEFAULT
+            # generate a randomized key so the player can choose a character name later
+            key = ''.join(choices(string.ascii_letters + string.digits, k=10))
+            new_character = create.create_object(_CHARACTER_TYPECLASS, key=key,
+                               location=None,
+                               home=default_home,
+                               permissions=permissions)
+            # only allow creator (and developers) to puppet this char
+            new_character.locks.add(
+                f"puppet:pid({account.id}) or perm(Developer) or pperm(Developer);delete:id({account.id}) or perm(Admin)"
+            )
+            # initalize the new character to the beginning of the chargen menu
+            new_character.db.chargen_step = "menunode_welcome"
+            account.db._playable_characters.append(new_character)
+
+        # set the menu node to start at to the character's last saved step
+        startnode = new_character.db.chargen_step
+        # attach the character to the session, so the chargen menu can access it
+        session.new_char = new_character
+
+        # this gets called every time the player exits the chargen menu
+        def finish_char_callback(session, menu):
+            char = session.new_char
+            if not char.db.chargen_step:
+                # this means character creation was completed - start playing!
+                # execute the ic command to start puppeting the character
+                account.execute_cmd("ic {}".format(char.key))
+
+        EvMenu(session,
+               settings.CHARGEN_MENU,
```

Comment:
I'd suggest loading this at the top, with an `AttributeError` leading to falling back to the example menu.